### PR TITLE
8344011: Remove usage of security manager from Class and reflective APIs

### PR DESCRIPTION
**Pull request by AlanBateman**
Work in progress.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344011](https://bugs.openjdk.org/browse/JDK-8344011): Remove usage of security manager from Class and reflective APIs (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22036/head:pull/22036` \
`$ git checkout pull/22036`

Update a local copy of the PR: \
`$ git checkout pull/22036` \
`$ git pull https://git.openjdk.org/jdk.git pull/22036/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22036`

View PR using the GUI difftool: \
`$ git pr show -t 22036`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22036.diff">https://git.openjdk.org/jdk/pull/22036.diff</a>

</details>

### CODE REVIEW

**Review by liach — COMMENTED** [review]
I see the removal of SM makes many jli implementation methods like `getXxxNoSecurityManager` obsolete.  Is the patch only aimed at direct interactions with the SM (i.e. terminally deprecated code), and these obsolete/unused methods are target for later cleanups?

Also there are a few PrivilegedAction missed, such as InnerClassLambdaMetafactory GetBooleanAction, MethodHandleStatics GetPropertyAction, ClassFileDumper comment about GetPropertyAction.  (I think those are all)

**Comment by AlanBateman** [comment]
> I see the removal of SM makes many jli implementation methods like `getXxxNoSecurityManager` obsolete. Is the patch only aimed at direct interactions with the SM (i.e. terminally deprecated code), and these obsolete/unused methods are target for later cleanups?

There will be a lot of PRs to remove usage of the SM from the libraries, the intention is to keep the changes as easy to review as possible.

File: src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -1991,36 +1868,18 @@ private boolean hasEnclosingMethodInfo() {
...
-                        for (Class<?> m : currentClass.getDeclaredClasses()) {
-                            if (Modifier.isPublic(m.getModifiers())) {
-                                list.add(m);
-                            }
-                        }
-                        currentClass = currentClass.getSuperclass();
-                    }
-                    return list.toArray(new Class<?>[0]);
+        List<Class<?>> list = new ArrayList<>();
+        Class<?> currentClass = Class.this;
```
[the remark this answers is not in the document]
Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/Class.java:1873]: The intention is to keep the changes as easy to review as possible.

**Comment by liach** [comment]
There is one standalone usage of SM in `jdk.internal.constant.MethodTypeDescImpl` of `AccessController.doPrivileged`.  Since it is maintained closely with invoke/reflect, can you update that site too?

**Comment by AlanBateman** [comment]
> There is one standalone usage of SM in `jdk.internal.constant.MethodTypeDescImpl` of `AccessController.doPrivileged`. Since it is maintained closely with invoke/reflect, can you update that site too?

It's in a different batch but I can move it to this one.

**Comment by liach** [comment]
Either way is fine; that change is tiny, and having it in a reflection-oriented or general-purposed patch makes no difference.

**Pull request closed without merging**